### DISCUSSION TITLE
How is Valibot different from arktype?

### DISCUSSION THREAD

**Discussion by asterikx** [open-circle/valibot#58] [Q&A]
Curious to know how Valibot compares to arktype. When would I use Valibot over arktype and vice versa? Thanks a lot 😊

**Comment by fabian-hiller** [comment]
I am currently writing my bachelor thesis and plan to investigate ArkType in it as well. I plan to publish the paper in September. Furthermore I will integrate the research into the Valibot docs.

**Reply by asterikx** [reply]
Great to hear! Good luck with your thesis 😁

**Comment by hyunbinseo** [comment]
Would love to have an update on this, especially since [ArkType v2 was released](https://x.com/arktypeio/status/1880328086023217644) on January 2025.

https://arktype.io/docs/intro/your-first-type

```ts
import { type } from 'arktype';

// define
const user = type({
  'name': 'string',
  'platform': "'android' | 'ios'",
  'versions?': '(number | string)[]',
});

/* {
  name: string;
  platform: 'android' | 'ios';
  versions?: (string | number)[] | undefined;
} */
type User = typeof user.infer;

// validate
const out = user({
  name: 'Alan Turing',
  device: {
    platform: 'enigma',
    versions: [0, '1', 0n],
  },
});

if (out instanceof type.errors) {
  // platform must be "android" or "ios" (was missing)
  console.error(out.summary);
} else {
  console.log(`Hello, ${out.name}`);
}
```

**Reply by fabian-hiller** [reply]
Unfortunately I have no time to write a detailed comparison. Valibot and ArkType are quite different. Both have some strengths and weaknesses. I think a lot of people love Valibot for its bundle size and API design, but if you haven't used a schema library like Zod before, ArkType might be easier to get started with because it tries to mirror the TS syntax.
